xcube_cds/datasets/drought_indices_era5.py

```diff
@@ -0,0 +1,233 @@
+from xcube.util.jsonschema import JsonStringSchema
+import zipfile
+
+from xcube_cds.store import CDSDatasetHandler
+
+
+class DroughtIndicesDatasetHandler(CDSDatasetHandler):
+
+    def __init__(self):
+        self._data_id_map = {
+            "derived-drought-historical-monthly:reanalysis": (
+                "Monthly drought indices from 1940–present derived "
+                "from ERA5 reanalysis (main run)"
+            ),
+        }
+        self._variable_names = [
+            "standardised_precipitation_index",
+            "standardised_precipitation_evapotranspiration_index",
+            "probability_of_zero_precipitation_spi",
+            "test_for_normality_spi",
+            "test_for_normality_spei",
+        ]
+        self._accumulation_periods = [1, 3, 6, 12, 24, 36, 48]
+        self._min_date = "1940-01-01"
+        self._max_date = "2025-12-31"
+        self._bbox = (-180.0, -90.0, 180.0, 90.0)
+        self._spatial_res = 0.25
+
+    def get_supported_data_ids(self) -> list[str]:
+        return list(self._data_id_map)
+
+    def get_open_data_params_schema(
+        self, data_id: str | None = None
+    ) -> JsonObjectSchema:
+        params = dict(
+            variable_names=JsonArraySchema(
+                items=(JsonStringSchema(enum=self._variable_names)),
+                min_items=1,
+                unique_items=True,
+                nullable=True,
+                description="identifiers of the requested variables",
+            ),
+            accumulation_periods=JsonArraySchema(
+                items=(JsonNumberSchema(enum=self._accumulation_periods)),
+                unique_items=True,
+                nullable=True,
+                description="Accumulation windows in months",
+            ),
+            bbox=JsonArraySchema(
+                items=(
+                    JsonNumberSchema(minimum=self._bbox[0], maximum=self._bbox[2]),
+                    JsonNumberSchema(minimum=self._bbox[1], maximum=self._bbox[3]),
+                    JsonNumberSchema(minimum=self._bbox[0], maximum=self._bbox[2]),
+                    JsonNumberSchema(minimum=self._bbox[1], maximum=self._bbox[3]),
+                ),
+                description="bounding box (min_x, min_y, max_x, max_y)",
+            ),
+            time_range=JsonDateSchema.new_range(
+                min_date=self._min_date, max_date=self._max_date
+            ),
+        )
+        required = [
+            "variable_names",
+            "accumulation_periods",
+            "bbox",
+            "time_range",
+        ]
+        return JsonObjectSchema(
+            properties=params, required=required, additional_properties=False
+        )
+
+    def get_human_readable_data_id(self, data_id: str) -> str:
+        return self._data_id_map[data_id]
+
+    def describe_data(self, data_id: str) -> DatasetDescriptor:
+        # ToDo add VariableDescriptors to data_vars
+        return DatasetDescriptor(
+            data_id=data_id,
```

Comment:
Please add a valid `data_vars` structure here.  You may find the `create_variable_descriptors_from_netcdf.py` script useful.